workers/data_refinery_workers/processors/smasher.py

```diff
@@ -491,379 +134,262 @@ def sync_quant_files(output_path, files_sample_tuple, job_context: Dict):
         latest_computed_file.get_synced_file_path(path=output_file_path)
     return num_samples
 
-def process_frame(inputs) -> Dict:
-    (job_context, computed_file, sample, index) = inputs
-    log_state('processing frame {}'.format(index), job_context["job"])
-    frame = {
-        "unsmashable": False,
-        "unsmashable_file": None,
-        "column_type": None,
-        "columns": None,
-        "data": None
-    }
-
-    def unsmashable(path):
-      frame['unsmashable'] = True
-      frame['unsmashable_file'] = path
-      return frame
-
-    def smashable(data):
-      frame['data'] = data
-      return frame
+def _inner_join(job_context: Dict) -> pd.DataFrame:
+    """Performs an inner join across the all_frames key of job_context.
 
-    try:
-        # Download the file to a job-specific location so it
-        # won't disappear while we're using it.
-        computed_file_path = job_context["work_dir"] + computed_file.filename
-        computed_file_path = computed_file.get_synced_file_path(path=computed_file_path)
-
-        # Bail appropriately if this isn't a real file.
-        if not computed_file_path or not os.path.exists(computed_file_path):
-            logger.warning("Smasher received non-existent file path.",
-                computed_file_path=computed_file_path,
-                computed_file=computed_file,
-                dataset_id=job_context['dataset'].id,
-            )
-            return unsmashable(computed_file_path)
+    Returns a new dict containing the metadata, not the job_context.
+    """
+    # Merge all of the frames we've gathered into a single big frame, skipping duplicates.
+    # TODO: If the very first frame is the wrong platform, are we boned?
+    # TODO: I think I'd like to not have all_frames be directly on the job context and overwritten.
+    # Hmm, now I'm less sure because I think that means we wouldn't overwrite them. Perhaps if I do in the
```

Comment:
is this cut off?

Reply:
Oh yeah... There's a few notes to myself I need to go remove. I do plan on leaving some TODOs in for subsequent refactor PRs.